File: BackEnd/backend/src/main/java/com/fitapp/backend/Exercise/aplication/dto/request/ExerciseFilterRequest.java
package com.fitapp.backend.Exercise.aplication.dto.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fitapp.backend.infrastructure.persistence.entity.enums.ExerciseType;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import org.springframework.data.domain.Sort;

import java.util.ArrayList;
import java.util.List;

@Data
@Schema(description = "Filtros para búsqueda de ejercicios")
public class ExerciseFilterRequest {
    
    @Schema(description = "Texto de búsqueda en nombre o descripción")
    @JsonProperty("search")
    private String search;
    
    @Schema(description = "Tipo de ejercicio")
    @JsonProperty("exerciseType")
    private ExerciseType exerciseType;
    
    @Schema(description = "ID del deporte")
    @JsonProperty("sportId")
    private Long sportId;
    
    @Schema(description = "ID de la categoría")
    @JsonProperty("categoryId")
    private Long categoryId;
    
    @Schema(description = "ID del parámetro soportado")
    @JsonProperty("parameterId")
    private Long parameterId;
    
    @Schema(description = "Si está activo")
    @JsonProperty("isActive")
    private Boolean isActive;
    
    @Schema(description = "Si es público")
    @JsonProperty("isPublic")
    private Boolean isPublic;
    
    @Schema(description = "ID del creador")
    @JsonProperty("createdBy")
    private Long createdBy;
    
    @Schema(description = "Incluir ejercicios públicos")
    @JsonProperty("includePublic")
    private Boolean includePublic = true;
    
    @Schema(description = "Mínimo rating")
    @JsonProperty("minRating")
    private Double minRating;
    
    @Schema(description = "Página solicitada (0-index)", example = "0")
    @JsonProperty("page")
    private Integer page = 0;
    
    @Schema(description = "Tamaño de página", example = "20")
    @JsonProperty("size")
    private Integer size = 20;
    
    @Schema(description = "Campo para ordenar", example = "name")
    @JsonProperty("sortBy")
    private String sortBy = "name";
    
    @Schema(description = "Dirección del orden", example = "ASC")
    @JsonProperty("direction")
    private Sort.Direction direction = Sort.Direction.ASC;
    
    @Schema(description = "Ordenar por popularidad (uso)")
    @JsonProperty("sortByPopularity")
    private Boolean sortByPopularity = false;
    
    @Schema(description = "Ordenar por rating")
    @JsonProperty("sortByRating")
    private Boolean sortByRating = false;
    
    @Schema(description = "Múltiples campos para ordenar")
    @JsonProperty("sortFields")
    private List<SortField> sortFields = new ArrayList<>();
    
    @Data
    public static class SortField {
        private String field;
        private Sort.Direction direction = Sort.Direction.ASC;
    }
}

File: FrontEnd/app/src/main/java/com/fitapp/appfit/feature/exercise/model/exercise/request/ExerciseFilterRequest.kt
package com.fitapp.appfit.feature.exercise.model.exercise.request

import com.fitapp.appfit.feature.exercise.model.exercise.response.ExerciseType
import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class ExerciseFilterRequest(
    @SerializedName("search")
    val search: String? = null,

    @SerializedName("exerciseType")
    val exerciseType: ExerciseType? = null,

    @SerializedName("sportId")
    val sportId: Long? = null,

    @SerializedName("categoryId")
    val categoryId: Long? = null,

    @SerializedName("parameterId")
    val parameterId: Long? = null,

    @SerializedName("isActive")
    val isActive: Boolean? = null,

    @SerializedName("isPublic")
    val isPublic: Boolean? = null,

    @SerializedName("createdBy")
    val createdBy: Long? = null,

    @SerializedName("includePublic")
    val includePublic: Boolean? = true,

    @SerializedName("minRating")
    val minRating: Double? = null,

    @SerializedName("page")
    val page: Int = 0,

    @SerializedName("size")
    val size: Int = 20,

    @SerializedName("sortBy")
    val sortBy: String = "name",

    @SerializedName("direction")
    val direction: SortDirection = SortDirection.ASC,

    @SerializedName("sortByPopularity")
    val sortByPopularity: Boolean = false,

    @SerializedName("sortByRating")
    val sortByRating: Boolean = false,

    @SerializedName("sortFields")
    val sortFields: List<SortField> = emptyList()
) : Serializable {
    enum class SortDirection {
        ASC,
        DESC
    }

    data class SortField(
        @SerializedName("field")
        val field: String,

        @SerializedName("direction")
        val direction: SortDirection = SortDirection.ASC
    ) : Serializable
}